src/python/WMCore/MicroService/Unified/MSTransferor.py

```diff
@@ -680,45 +650,143 @@ def makeTransferRequest(self, wflow):
+        :param dataIn: short summary of the data to be placed
+        :param subLevel: subscription level (dataset/container or block)
+        :param blocks: list of blocks to be subscribed (or None if dataset level)
+        :param dataSize: amount of data being placed by this rule
+        :param nodes: list of nodes/RSE
+        :param nodeIdx: index of the node/RSE to be used in the replication rule
+        :return: a boolean flagging whether it succeeded or not, and the rule id
+        """
+        subLevel = "ALL" if subLevel == "dataset" else "DATASET"
+        dids = blocks if blocks else dataIn['name']
+        rseExpr = nodes[nodeIdx]
+        rule = {'copies': 1,
+                'ask_approval': True if self.msConfig["phedexRequestOnly"] == "y" else False,
+                'activity': 'Production Input',
+                'account': self.msConfig['rucioAccount'],
+                'grouping': subLevel,
+                'comment': 'WMCore MSTransferor input data placement'}
+        msg = "Creating '{}' level subscription for {} dataset: {}".format(subLevel,
+                                                                           dataIn['type'],
+                                                                           dataIn['name'])
+        if wflow.getParentDataset():
+            msg += ", where parent blocks have also been added for dataset: %s" % wflow.getParentDataset()
+        self.logger.info(msg)
+
         success, transferId = True, 0
-        try:
-            if hasattr(self, "rucio"):
-                # FIXME: then it should create a Rucio rule instead
-                res = self.phedex.subscribe(subscriptionObj)
-            else:
-                res = self.phedex.subscribe(subscriptionObj)
-            self.logger.debug("Subscription done, result: %s", res)
-        except HTTPException as ex:
-            # It might be that the block has no more replicas (all files invalidated)
-            # let it go and fail at global workqueue level
-            msg = "Subscription failed for workflow: %s and dataset: %s " % (wflowName, dsetName)
-            if ex.status == 400 and "request matched no data in TMDB" in ex.reason:
-                msg += "because data cannot be found in TMDB. Bypassing this/these blocks..."
-                self.logger.warning(msg)
+        if self.msConfig.get('enableDataTransfer', True):
+            # Force request-only subscription
+            # to any data transfer going above some threshold (do not auto-approve)
+            aboveWarningThreshold = self.msConfig.get('warningTransferThreshold') > 0. and \
+                                    dataSize > self.msConfig.get('warningTransferThreshold')
+            if aboveWarningThreshold and rule['ask_approval'] is False:
+                rule['ask_approval'] = True
+
+            # Then make the data subscription, for real!!!
+            res = self.rucio.createReplicationRule(dids, rseExpr, **rule)
```

Comment:
It's probably a better idea indeed. Especially because I am not sure whether this AccessDenied is a general issue related to the account/credentials we are using (this is what I assumed in the first place), or whether it could be related to RSE's that we cannot write to (or approve things)... I'll update it too